code/game/objects/items/toys.dm

```diff
@@ -770,26 +770,29 @@
 		return
 	var/choice = null
 	if(cards.len == 0)
-		src.icon_state = "deck_[deckstyle]_empty"
 		user << "<span class='warning'>There are no more cards to draw!</span>"
 		return
 	var/obj/item/toy/cards/singlecard/H = new/obj/item/toy/cards/singlecard(user.loc)
 	choice = cards[1]
 	H.cardname = choice
 	H.parentdeck = src
-	H.add_fingerprint(user)
 	var/O = src
 	H.apply_card_vars(H,O)
 	src.cards -= choice
 	H.pickup(user)
-	user.put_in_active_hand(H)
+	user.put_in_hands(H)
```

Comment:
put_in_active_hand drops the item if active hand is full, when put_in_hands check if the other hand is free first.